src/lib/generateRules.js

```diff
@@ -152,7 +152,27 @@ function applyVariant(variant, matches, context) {
 
   // Retrieve "modifier"
   {
-    let match = /(.*)\/(.*)$/g.exec(variant)
+    let start = null
+    let parens = 0
+    let brackets = 0
+
+    for (let i = 0; i < variant.length; i++) {
```

Comment:
I was wondering if we could use the `splitAtTopLevelOnly` function that we already have 🤔

Reply:
That's a good idea but it doesn't work because of this example: `ar-1/10/20:text-red-500`. I'll see if there's a way I can massage it into working though.

Reply:
Okay, I've tweaked the implementation to use `splitAtTopLevelOnly` _and_ account for it. It doesn't feel pretty.

It'd be nice if there was an option like a `limit` but where it could also accept negative numbers for N occurrences from the end kinda like `slice`. I'm not sure if that's feasible though. Or even the right API.

Reply:
Oh I see, yeah I noticed you had a `2` in the call but that's not used by splitAtTopLvelOnly afaik 🤔

Reply:
oh crap I left that idea in lol. It didn't pan out hahaha.